test-none-dynamic-host 运行时崩溃

`
2022-01-20 16:08:00.784 12230-12230/com.tencent.shadow.test.hostapp E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.tencent.shadow.test.hostapp, PID: 12230
    java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/test/espresso/IdlingResource;
        at dalvik.system.DexFile.defineClassNative(Native Method)
        at dalvik.system.DexFile.defineClass(DexFile.java:294)
        at dalvik.system.DexFile.loadClassBinaryName(DexFile.java:287)
        at dalvik.system.DexPathList$Element.findClass(DexPathList.java:773)
        at dalvik.system.DexPathList.findClass(DexPathList.java:533)
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:205)
        at com.tencent.shadow.core.loader.classloaders.PluginClassLoader.loadClass(PluginClassLoader.kt:101)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at java.lang.Class.classForName(Native Method)
        at java.lang.Class.forName(Class.java:454)
        at java.lang.Class.forName(Class.java:379)
        at com.tencent.shadow.test.plugin.general_cases.lib.usecases.activity.TestListActivity$1.onItemClick(TestListActivity.java:39)
        at android.widget.AdapterView.performItemClick(AdapterView.java:330)
        at android.widget.AbsListView.performItemClick(AbsListView.java:1187)
        at android.widget.AbsListView$PerformClick.run(AbsListView.java:3179)
        at android.widget.AbsListView.onTouchUp(AbsListView.java:4109)
        at android.widget.AbsListView.onTouchEvent(AbsListView.java:3867)
        at android.view.View.dispatchTouchEvent(View.java:14309)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3112)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2785)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3118)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2799)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3118)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2799)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3118)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2799)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3118)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2799)
        at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:488)
        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1871)
        at android.app.Activity.dispatchTouchEvent(Activity.java:4125)
        at com.tencent.shadow.core.runtime.container.GeneratedPluginContainerActivity.superDispatchTouchEvent(GeneratedPluginContainerActivity.java:1550)
        at com.tencent.shadow.core.runtime.container.PluginContainerActivity.superDispatchTouchEvent(PluginContainerActivity.java:38)
        at com.tencent.shadow.core.runtime.GeneratedPluginActivity.dispatchTouchEvent(GeneratedPluginActivity.java:467)
        at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:446)
        at android.view.View.dispatchPointerEvent(View.java:14568)
        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:6016)
        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5819)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5310)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5367)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5333)
        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5485)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5341)
        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5542)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5314)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5367)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5333)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5341)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5314)
        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:8080)
2022-01-20 16:08:00.785 12230-12230/com.tencent.shadow.test.hostapp E/AndroidRuntime:     at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:8031)
        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7992)
        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:8203)
        at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:220)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:335)
        at android.os.Looper.loop(Looper.java:183)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "androidx.test.espresso.IdlingResource" on path: DexPathList[[zip file "/data/user/0/com.tencent.shadow.test.hostapp/files/plugin.apk"],nativeLibraryDirectories=[/data/user/0/com.tencent.shadow.test.hostapp/files/plugin.apk_lib, /system/lib, /system_ext/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:207)
        at com.tencent.shadow.core.loader.classloaders.PluginClassLoader.loadClass(PluginClassLoader.kt:101)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        	... 61 more
    	Suppressed: java.lang.ClassNotFoundException: androidx.test.espresso.IdlingResource
        at java.lang.Class.classForName(Native Method)
        at java.lang.BootClassLoader.findClass(ClassLoader.java:1358)
        at java.lang.BootClassLoader.loadClass(ClassLoader.java:1418)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.tencent.shadow.core.loader.classloaders.PluginClassLoader.loadClass(PluginClassLoader.kt:95)
        		... 62 more
     Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
`

--- a/projects/test/none-dynamic/host/test-none-dynamic-host/build.gradle
+++ b/projects/test/none-dynamic/host/test-none-dynamic-host/build.gradle
@@ -47,6 +47,7 @@ dependencies {
     implementation 'com.tencent.shadow.core:loader'
     implementation 'com.tencent.shadow.core:activity-container'
     implementation project(':custom-view')
+    implementation 'androidx.test.espresso:espresso-idling-resource:3.3.0'
 }
 
 def createCopyTask(projectName, buildType, name, apkName) {

--- a/projects/test/none-dynamic/host/test-none-dynamic-host/src/main/java/com/tencent/shadow/test/none_dynamic/host/HostApplication.java
+++ b/projects/test/none-dynamic/host/test-none-dynamic-host/src/main/java/com/tencent/shadow/test/none_dynamic/host/HostApplication.java
@@ -28,16 +28,13 @@ import com.tencent.shadow.core.common.InstalledApk;
 import com.tencent.shadow.core.common.LoggerFactory;
 import com.tencent.shadow.core.load_parameters.LoadParameters;
 import com.tencent.shadow.core.loader.ShadowPluginLoader;
-import com.tencent.shadow.core.loader.exceptions.LoadPluginException;
 import com.tencent.shadow.core.runtime.container.ContentProviderDelegateProviderHolder;
 import com.tencent.shadow.core.runtime.container.DelegateProviderHolder;
 
 import java.util.HashMap;
 import java.util.Map;
-import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
 
 public class HostApplication extends Application {
     private static Application sApp;
@@ -66,7 +63,15 @@ public class HostApplication extends Application {
         }
 
         if (mPluginLoader.getPluginParts(partKey) == null) {
-            LoadParameters loadParameters = new LoadParameters(null, partKey, null, null);
+            // 插件访问宿主类的白名单
+            String[] hostWhiteList = new String[]{
+                    "androidx.test.espresso",//这个包添加是为了general-cases插件中可以访问测试框架的类
+                    "com.tencent.shadow.test.lib.plugin_use_host_code_lib.interfaces"//测试插件访问宿主白名单类
+            };
+            LoadParameters loadParameters = new LoadParameters(null,
+                    partKey,
+                    null,
+                    hostWhiteList);
 
             Parcel parcel = Parcel.obtain();
             loadParameters.writeToParcel(parcel, 0);

